core/crypto/src/signature.rs

```diff
@@ -556,6 +558,80 @@ impl<'de> serde::Deserialize<'de> for SecretKey {
 #[derive(Clone)]
 pub struct Secp256K1Signature([u8; 65]);
 
+impl Secp256K1Signature {
+    fn check_v(&self) -> Result<u8, crate::errors::ParseSignatureError> {
+        match self.0[64] {
+            27 | 28 => Ok(self.0[64] - 27),
+            _ => Err(crate::errors::ParseSignatureError::InvalidData {
+                error_message: "invalid `v` value".to_string(),
+            }),
+        }
+    }
+
+    pub fn check_signature_values(&self) -> bool {
+        let mut r_bytes = [0u8; 32];
+        r_bytes.copy_from_slice(&self.0[0..32]);
+        let r = U256::from(r_bytes);
+
+        let mut s_bytes = [0u8; 32];
+        s_bytes.copy_from_slice(&self.0[32..64]);
+        let s = U256::from(s_bytes);
+
+        let _v = match self.check_v() {
+            Ok(v) => v,
+            Err(_e) => return false,
+        };
```

Comment:
```suggestion
        if let Err(_) = match self.check_v() {
            return false;
        };
```

Reply:
```suggestion
        if match self.check_v().is_err() {
            return false;
        };
```

Same as above, but no need for the let binding

Reply:
No longer required as this functionality was removed. https://github.com/near/nearcore/pull/4380/commits/1f2a3141f83913957822ab45e4f421405519871f